Sources/DeploymentTargetAWSLambda/AWSStuff.swift

```diff
@@ -0,0 +1,473 @@
+//
+//  File.swift
+//  
+//
+//  Created by Lukas Kollmer on 2021-01-19.
+//
+
+import Foundation
+import ApodiniUtils
+import class Apodini.AnyHandlerIdentifier
+import ApodiniDeployBuildSupport
+import DeploymentTargetAWSLambdaCommon
+import Logging
+import NIO
+import SotoLambda
+import SotoApiGatewayV2
+import SotoS3
+import SotoS3FileTransfer
+import SotoIAM
+import SotoSTS
+import OpenAPIKit
+
+
+
+
+/// A type which interacts with AWS to create and configure ressources.
+/// - Note: Instances of this class should not be re-used to apply multiple deployments.
+class AWSDeploymentStuff { // needs a better name
+    private static let lambdaFunctionNamePrefix = "apodini-lambda"
+    private static let xAmazonApigatewayIntegrationKey = "x-amazon-apigateway-integration"
+    
+    private let tmpDirUrl: URL
+    private let FM = FileManager.default
+    private let logger = Logger(label: "de.lukaskollmer.ApodiniLambda.AWSIntegration")
+    
+    private let awsProfileName: String
+    private let awsRegion: SotoCore.Region
+    private let awsClient: AWSClient
+    private let sts: STS
+    private let iam: IAM
+    private let s3: S3
+    private let lambda: Lambda
+    private let apiGateway: ApiGatewayV2
+    
+    private var didRunDeployment = false
+    private var lambdaExecutionRole: IAM.Role?
+    private var deployedLambdaFunctions: Set<String> = [] // Set of lambda function names
+    
+    
+    init(
+        awsProfileName: String,
+        awsRegionName: String,
+        tmpDirUrl: URL
+    ) {
+        self.awsProfileName = awsProfileName
+        self.awsRegion = .init(rawValue: awsRegionName)
+        self.tmpDirUrl = tmpDirUrl
+        awsClient = AWSClient(
+            credentialProvider: .configFile(profile: awsProfileName),
+            retryPolicy: .exponential(),
+            httpClientProvider: .createNew
+        )
```

Comment:
As we no longer rely on the CLI we might also think about adopting some other credential provider techniques that soto offers. I would imagine a command linke argument that configures how the credentials are loaded.

Reply:
Yes that's one of the items on my to-do list, as part of the AWS ressource mgmt thing.
My current idea is to make the --aws-profile-name optional (which i think technically speaking it already is), and then try to detect AWS credentials dynamically:
- If the CLI parameter is defined, that value always takes precedence and we try to use that profile in the config file
- If the CLI parameter is not defined, and we find `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` environment variables, we use these
- If the CLI parameter is not defined, and there are no environment variables, we try to use the "default" profile from the config file

Reply:
That sounds like a good idea 🚀